### DISCUSSION TITLE
How to properly store videos for notes?

### DISCUSSION THREAD

**Discussion by DuckersMcQuack** [TriliumNext/Trilium#6000] [Ideas]
I want to put videos in notes, several which can exeed 250MB. Is there a way to store those in my regular documents share on unraid, and have trilium read from those to embed and play in the notes?

**Comment by perfectra1n** [comment]
My first thought would be to store them somewhere more apt like Nextcloud, and then use a Trilium web view note type to render/play the video using the "share" link from Nextcloud. Storing videos in an SQLite DB is a really, really bad idea. Same goes for any SQL DB - you'd want the actual video file to live on a filesystem.

**Reply by perfectra1n** [reply]
And perhaps the "document share" on Unraid is precisely that, which would replace the need for Nextcloud.

**Reply by DuckersMcQuack** [reply]
> My first thought would be to store them somewhere more apt like Nextcloud, and then use a Trilium web view note type to render/play the video using the "share" link from Nextcloud. Storing videos in an SQLite DB is a really, really bad idea. Same goes for any SQL DB - you'd want the actual video file to live on a filesystem.

Aye. That's what i want.  Have the videos in a share on my unraid server for instance, then a addon/plugin for trilium to read those videos from said share that you set up in it's docker configs. Where no videos are stored in trilium itself, heck, even same with images and other files not being text, just have trilium read from that folder containing files. Where when you upload images/videos to the notes, it won't store those medias/files in it's appdata, but in it's designated folder on a share.

Would that work?

**Reply by perfectra1n** [reply]
If you'd want to use the Web View note type, you would need to use some service that would serve the video/image files over a web-based service, such as Nextcloud. I can't think of a way to render a file that's on a remote SMB share within Trilium, unless you cook something up yourself.

**Reply by DuckersMcQuack** [reply]
Ah, so it's unable to read from videos on simply just a share/folder in a share? It needs to be webview reading from say a locally hosted "file host service" docker?

**Reply by kalmigs** [reply]
Yeah it would be good if we can have a shared folder. so pictures, videos or even any kind of attachments can be saved there and referenced through out the app
